### ISSUE TITLE
Put's without a key

### ISSUE THREAD

**Issue by csilvers** [Khan/kotlin-datastore#10]
It seems like subclasses of Keyable<> need to have a non-null key.  But if you're creating a new entity to put(), it might have -- or you might want it to have -- a null key, and let the datastore assign a key.  Is there a way to do that in kotlin-datastore?

**Comment by cjfuller** [comment]
No, I don't believe we have that functionality at present. (You could try providing a numeric key ID of 0 and seeing if that works? That would act as the default "no key provided" value in the protobuf, but it's also possible that the java client library will refuse that value.)

**Comment by kairuiwang** [comment]
Hi @cjfuller !! Thanks for responding! I tried to provide 0 as a default key ID and at runtime, I did get an error saying something to the effect of `Id must not equal zero`. 😢
